src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5851,6 +5843,8 @@ private VariableState GetVariableState(Optional<LocalState> localState)
                 refKinds.AddRange(argumentRefKindsOpt);
             }
 
+            var binder = _binder ?? compilation.GetBinder(callSyntax);
```

Comment:
@AlekseyTs this is the location I was talking about yesterday where I managed to get the binder usage down to just one call: http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Binder/Semantics/OverloadResolution/MethodTypeInference.cs,1334 (the rest of the usages in the chain can be replaced by passing the nested thing on the binder it's accessing). Unfortunately, that last call doesn't seem to be easily fixable. The only time `_binder` should be null is when we're binding the body of an unbound lambda, trying to figure out the return type of that lambda. I'm not certain getting the binder from the compilation is the right thing to do here, particularly around speculation and potentially getting the wrong binder for such cases. Do you have any thoughts?

I still have a bit of other work to do on this PR around using/foreach dispose argument binding, so no rush on this question.

Reply:
> The only time _binder should be null is when we're binding the body of an unbound lambda, trying to figure out the return type of that lambda. I'm not certain getting the binder from the compilation is the right thing to do here, particularly around speculation and potentially getting the wrong binder for such cases. Do you have any thoughts?

I believe an UnboundLambda itself holds a reference to a Binder, indirectly. Can we use that binder? 

---
In reply to: [547419319](https://github.com/dotnet/roslyn/pull/50105#discussion_r547419319) [](ancestors = 547419319)

Reply:
Looks like this an option I missed, thanks.

---
In reply to: [548001479](https://github.com/dotnet/roslyn/pull/50105#discussion_r548001479) [](ancestors = 548001479,547419319)